npm/asciidoctor-revealjs

```diff
@@ -0,0 +1,130 @@
+#!/usr/bin/env node
+
+'use strict'
+
+process.title = 'asciidoctor-revealjs'
+
+const pkg = require('../package.json')
+const { Options, Invoker, processor } = require('@asciidoctor/cli')
+const asciidoctor = require('@asciidoctor/core')();
+const asciidoctorRevealjs = require('@asciidoctor/reveal.js')
+const path = require('path')
+const EventEmitter = require('events')
+class ProcessorEmitter extends EventEmitter {}
+const processorEmitter = new ProcessorEmitter()
+
+async function convertFiles (files, argv, options, verbose) {
+  for (const file of files) {
+    if (verbose) {
+      console.log(`converting file ${file}`)
+    }
+    //await converter.convert(processor, file, options, argv.timings)
+    options['backend'] = "revealjs"
+    await asciidoctor.convert(file, options, argv.timings)
+  }
+}
+
+let processing = false
+let shouldEmitRequestEvent = false
+processorEmitter.on('request', async (files, args, options, verbose, preview) => {
+  try {
+    // prevent multiprocessing.
+    if (!processing) {
+      processing = true
+      try {
+        await convertFiles(files, args, options, verbose, preview)
+      } finally {
+        processing = false
+        if (shouldEmitRequestEvent) {
+          shouldEmitRequestEvent = false
+          processorEmitter.emit('request', files, args, options, verbose, preview)
+        }
+      }
+    } else {
+      shouldEmitRequestEvent = true
+    }
+  } catch (e) {
+    console.log('error', e)
+  }
+})
+
+class RevealJsInvoker extends Invoker {
+  async invoke () {
+    const processArgs = this.options.argv.slice(2)
+    const { args } = this.options
+    const { verbose, version, files } = args
+
+    if (version || (verbose && processArgs.length === 1)) {
+      process.stdout.write(`Asciidoctor-Reveal.js ${pkg.version} using `)
```

Comment:
I think I need to define `printVersion` as an instance function (instead of a static function).

```js
class RevealJsInvoker extends Invoker {

  printVersion () {
    process.stdout.write(`Asciidoctor-Reveal.js ${pkg.version} using `)
    super.printVersion()
  }
}

Reply:
I can confirm. I added the code but it won't work until you call this dynamically: https://github.com/asciidoctor/asciidoctor-cli.js/blob/master/lib/invoker.js#L20.

Do you consider this a big change for the `@asciidoctor/cli` or are you willing to do it in a point or feature release?

Reply:
> Do you consider this a big change for the @asciidoctor/cli or are you willing to do it in a point or feature release?

I think I can ensure backward compatibility but I might need to use another name for the function.